server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterGetSettingsAction.java

```diff
@@ -50,20 +61,55 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        ClusterStateRequest clusterStateRequest = Requests.clusterStateRequest().routingTable(false).nodes(false);
         final boolean renderDefaults = request.paramAsBoolean("include_defaults", false);
-        clusterStateRequest.local(request.paramAsBoolean("local", clusterStateRequest.local()));
-        clusterStateRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterStateRequest.masterNodeTimeout()));
+
+        if (nodesInCluster.get().getMinNodeVersion().onOrAfter(Version.V_8_3_0) == false) {
+            return prepareLegacyRequest(request, client, renderDefaults);
+        }
+
+        ClusterGetSettingsAction.Request clusterSettingsRequest = new ClusterGetSettingsAction.Request();
+
+        clusterSettingsRequest.local(request.paramAsBoolean("local", clusterSettingsRequest.local()));
+        clusterSettingsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterSettingsRequest.masterNodeTimeout()));
+
         return channel -> client.admin()
             .cluster()
-            .state(
-                clusterStateRequest,
+            .clusterSettings(
+                clusterSettingsRequest,
                 new RestToXContentListener<RestClusterGetSettingsResponse>(channel).map(
-                    response -> response(response.getState(), renderDefaults, settingsFilter, clusterSettings, settings)
+                    response -> response(
+                        response.persistentSettings(),
+                        response.transientSettings(),
+                        response.settings(),
+                        renderDefaults,
+                        settingsFilter,
+                        clusterSettings,
+                        settings
+                    )
                 )
             );
     }
 
+    private RestChannelConsumer prepareLegacyRequest(final RestRequest request, final NodeClient client, final boolean renderDefaults) {
+        ClusterStateRequest clusterStateRequest = Requests.clusterStateRequest().routingTable(false).nodes(false);
+        clusterStateRequest.local(request.paramAsBoolean("local", clusterStateRequest.local()));
+        clusterStateRequest.masterNodeTimeout(request.paramAsTime("master_timeout", clusterStateRequest.masterNodeTimeout()));
+        return channel -> client.admin()
+            .cluster()
+            .state(clusterStateRequest, new RestToXContentListener<RestClusterGetSettingsResponse>(channel).map(response -> {
```

Comment:
I think it would be slightly nicer to use the same `ActionListener<ClusterGetSettingsAction.Response>` for both the legacy and the regular case, with a `.map(clusterState -> new ClusterGetSettingsAction.Response(...))` to do the extra conversion step in the legacy case.